components/Dialogs/src/private/MDCAlertControllerView+Private.m

```diff
@@ -0,0 +1,597 @@
+  // TODO(#1726): Determine default text color values for Normal and Disabled
+  CGRect buttonRect = actionButton.bounds;
+  buttonRect.size.height = MAX(buttonRect.size.height, MDCDialogActionButtonHeight);
+  buttonRect.size.width = MAX(buttonRect.size.width, MDCDialogActionButtonMinimumWidth);
+  actionButton.frame = buttonRect;
+  [actionButton addTarget:nil
+                   action:selector
+         forControlEvents:UIControlEventTouchUpInside];
+  [self.actionsScrollView addSubview:actionButton];
+
+  [_actionButtons addObject:actionButton];
+}
+
+- (void)setTitleFont:(UIFont *)font {
+  _titleFont = font;
+
+  [self updateTitleFont];
+}
+
+
+- (void)updateTitleFont {
+  // If we have a custom font apply it to the label.
+  // If not, fall back to the Material specified font.
+  if (_titleFont) {
+    // If we are automatically adjusting for Dynamic Type resize the font based on the text style
+    if (_mdc_adjustsFontForContentSizeCategory) {
+      _titleLabel.font =
+      [_titleFont mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
+                               scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+    } else {
+      _titleLabel.font = _titleFont;
+    }
+  } else {
+    // TODO(#2709): Migrate to a single source of truth for fonts
+    // There is no custom font, so use the default font.
+    if (_mdc_adjustsFontForContentSizeCategory) {
+      // If we are using the default (system) font loader, retrieve the
+      // font from the UIFont preferredFont API.
+      if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+        _titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:kTitleTextStyle];
+      } else {
+        // There is a custom font loader, retrieve the font and scale it.
+        UIFont *customTypographyFont = [MDCTypography titleFont];
+        _titleLabel.font =
+        [customTypographyFont mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
+                                           scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+      }
+    } else {
+      // If we are using the default (system) font loader, retrieve the
+      // font from the UIFont standardFont API.
+      if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+        _titleLabel.font = [UIFont mdc_standardFontForMaterialTextStyle:kTitleTextStyle];
+      } else {
+        // There is a custom font loader, retrieve the font from it.
+        _titleLabel.font = [MDCTypography titleFont];
+      }
+    }
+  }
+
+  [self setNeedsLayout];
+}
+
+- (void)setTitleColor:(UIColor *)titleColor {
+  _titleColor = titleColor;
+
+  _titleLabel.textColor = titleColor;
+}
+
+- (NSString *)message {
+  return self.messageLabel.text;
+}
+
+- (void)setMessage:(NSString *)message {
+  self.messageLabel.text = message;
+
+//  self.preferredContentSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
+
+  [self setNeedsLayout];
```

Comment:
do we need this here and in the VC? Can this be called directly?

Reply:
We do not.  Removed from VC.